docs/troubleshooting.md

```diff
@@ -0,0 +1,83 @@
+# Troubleshooting
+
+Check if you are not hitting one of the issues listed below.
+
+## Handling of Assembly version Conflicts
+
+OpenTelemetry SDK NuGet package are deployed with the OpenTelemetry .NET Instrumentation.
+Conflicts in assemblies referenced by "source instrumentations", should be handled in
+the same way: updating the project references to ensure that they are the same version
+as the one used by the instrumentation.
+
+However, the workarounds proposed above only work at build time. When a rebuild is not
+possible, there are ways to force the application to use the assembly versions shipped
+together with the instrumentation.
+
+For .NET Framework applications, the workaround is to use Binding Redirects. For .NET Core
+[Framework-dependent deployment](https://docs.microsoft.com/en-us/dotnet/core/deploying/deploy-with-vs?tabs=vs156#framework-dependent-deployment)
+applications,
+[Additional-deps](https://github.com/dotnet/runtime/blob/main/docs/design/features/additional-deps.md)
+and [runtime package store](https://docs.microsoft.com/en-us/dotnet/core/deploying/runtime-store)
+from OpenTelemetry .NET Auto-Instrumentation installation location can be used as workaround.
+For other .NET Core deployment models, the workaround is to manipulate the `deps.json`. Currently,
+we do not automate any of these workarounds, but we are manually validating them.
```

Comment:
```suggestion
For other .NET Core deployment models, edit the `deps.json` file.
```

I wouldn't mention the lack of automation in Troubleshooting, but move it to a development or contributing guide (or even a TO-DO).